flink-runtime/src/main/java/org/apache/flink/runtime/jobmanager/slots/TaskManagerGateway.java

```diff
@@ -186,4 +187,12 @@ void triggerCheckpoint(
 	 * @return Future blob key under which the task manager stdout file has been stored
 	 */
 	CompletableFuture<TransientBlobKey> requestTaskManagerStdout(final Time timeout);
+
+	/**
+	 * Notify the slot is unused.
+	 *
+	 * @param allocationID allocation id of the slot
+	 * @return Future acknowledge if notify successfully
+	 */
+	CompletableFuture<Acknowledge> notifySlotUnused(AllocationID allocationID);
```

Comment:
The `TaskExecutor` contains the method `freeSlot(AllocationID allocationId, Throwable cause)`. Maybe we should simply expose this method as a RPC call. Then we don't have to implement a new method which simply calls `freeSlot`.